File: Android/SampleCode/app/src/main/java/com/doinglab/sdk/example/foodlenstestapplication/listview/ListViewAdapter.java
package com.doinglab.sdk.example.foodlenstestapplication.listview;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.doinglab.sdk.example.foodlenstestapplication.R;

import java.util.ArrayList;

public class ListViewAdapter extends BaseAdapter {
    private ArrayList<ListViewItem> listViewItemList = new ArrayList<ListViewItem>() ;

    public ListViewAdapter() {

    }

    public void clearItems()
    {
        listViewItemList.clear();
    }

    @Override
    public int getCount() {
        return listViewItemList.size() ;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final int pos = position;
        final Context context = parent.getContext();

        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.list_item, parent, false);
        }

        ImageView iconImageView = (ImageView) convertView.findViewById(R.id.img_food) ;
        TextView tv_foodname = (TextView) convertView.findViewById(R.id.tv_foodname) ;
        TextView tv_food_nutrition_info = (TextView) convertView.findViewById(R.id.tv_food_nutrition_info) ;
        TextView tv_food_position = (TextView) convertView.findViewById(R.id.tv_food_position) ;


        ListViewItem listViewItem = listViewItemList.get(position);

        iconImageView.setImageDrawable(listViewItem.getIcon());
        tv_foodname.setText(listViewItem.getTitle());
        tv_food_nutrition_info.setText(listViewItem.getFoodNutrition());
        tv_food_position.setText(listViewItem.getFoodPosition());

        return convertView;
    }

    @Override
    public long getItemId(int position) {
        return position ;
    }

    @Override
    public Object getItem(int position) {
        return listViewItemList.get(position) ;
    }

    public void addItem(Drawable icon, String title, String nutritonInfo, String boxLoc) {
        ListViewItem item = new ListViewItem();

        if(icon != null)
            item.setIcon(icon);


        if(title != "")
            item.setTitle(title);
        else
            item.setTitle("");

        if(nutritonInfo != "")
            item.setFoodNutrition(nutritonInfo);
        else
            item.setFoodNutrition("");

        if(title != "")
            item.setFoodPosition(boxLoc);
        else
            item.setFoodPosition("");

        listViewItemList.add(item);
    }
}

File: Android/SampleCode_Kotlin/app/src/main/java/com/doinglab/sdk/example/foodlenstestapplication/listview/ListViewAdapter.kt
package com.doinglab.sdk.example.foodlenstestapplication.listview

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.doinglab.sdk.example.foodlenstestapplication.R

class ListViewAdapter : BaseAdapter() {
    val listViewItemList = ArrayList<ListViewItem>()

    fun clearItems() {
        listViewItemList.clear()
    }

    override fun getCount(): Int {
        return listViewItemList.size
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var convertView = convertView
        val pos = position
        val context = parent.context

        if (convertView == null) {
            val inflater =
                context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            convertView = inflater.inflate(R.layout.list_item, parent, false)
        }

        val iconImageView = convertView!!.findViewById<View>(R.id.img_food) as ImageView
        val tv_foodname = convertView!!.findViewById<View>(R.id.tv_foodname) as TextView
        val tv_food_nutrition_info =
            convertView!!.findViewById<View>(R.id.tv_food_nutrition_info) as TextView
        val tv_food_position = convertView!!.findViewById<View>(R.id.tv_food_position) as TextView
        val listViewItem = listViewItemList[position]

        iconImageView.setImageDrawable(listViewItem.iconDrawable)
        tv_foodname.text = listViewItem.titleStr
        tv_food_nutrition_info.text = listViewItem.foodNutrition
        tv_food_position.text = listViewItem.foodPosition

        return convertView
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItem(position: Int): Any? {
        return listViewItemList[position]
    }

    fun addItem(icon: Drawable?, title: String, nutritonInfo: String, boxLoc: String?) {
        val item = ListViewItem()
        if (icon != null)
            item.iconDrawable = icon

        if (title !== "")
            item.titleStr = title
        else
            item.titleStr = ""

        if (nutritonInfo !== "")
            item.foodNutrition = nutritonInfo
        else
            item.foodNutrition = ""

        if (title !== "")
            item.foodPosition = boxLoc
        else
            item.foodPosition = ""

        listViewItemList.add(item)
    }
}